source/lac/trilinos_tpetra_sparse_matrix.cc

```diff
@@ -0,0 +1,46 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+
+#  include <deal.II/lac/trilinos_tpetra_sparse_matrix.templates.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+#  ifndef DOXYGEN
+// explicit instantiations
+namespace LinearAlgebra
+{
+  namespace TpetraWrappers
+  {
+    template class SparseMatrix<double>;
+
+    template void
+    SparseMatrix<double>::reinit(
+      const IndexSet                       &row_parallel_partitioning,
+      const IndexSet                       &col_parallel_partitioning,
+      const dealii::DynamicSparsityPattern &sparsity_pattern,
+      const MPI_Comm                        communicator,
+      const bool                            exchange_data);
+
+  } // namespace TpetraWrappers
+} // namespace LinearAlgebra
```

Comment:
This is probably the right thing for a start, but do we eventually want to instantiate also for other scalar types?

Reply:
See [my branch](https://github.com/jpthiele/dealii/tree/tpetra_wrappers_inst) where I already started instatiating everything for all the supported scalar types, i.e. depending on the actual Trilinos installation that is linked to deal.II.

I will test and update that branch once this PR is merged to stay consistent.

Reply:
I mark this as resolved since @jpthiele will work on this later.